pkg/resource/user_group/custom_update_api.go

```diff
@@ -0,0 +1,182 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package user_group
+
+import (
+	"context"
+	"errors"
+	svcapitypes "github.com/aws-controllers-k8s/elasticache-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	"github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	svcsdk "github.com/aws/aws-sdk-go/service/elasticache"
+)
+
+// Implements custom logic for UpdateUserGroup
+func (rm *resourceManager) customUpdateUserGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+
+	// Check for the user group status
+	if latest.ko.Status.Status == nil || *latest.ko.Status.Status != "active" {
+		return nil, requeue.NeededAfter(
+			errors.New("user group can not be modified, it is not in 'active' state"),
+			requeue.DefaultRequeueAfterDuration)
+	}
```

Comment:
Also include `PendingChanges` check such that if there are pending changes, then defer this update call.